src/ip_source.h

```diff
@@ -0,0 +1,56 @@
+/*****************************************************************************
+ * This file is part of libmicrodns.
+ *
+ * Copyright © 2014-2016 VideoLabs SAS
+ *
+ * Author: Jonathan Calmels <jbjcalmels@gmail.com>
+ *
+ *****************************************************************************
+ * libmicrodns is released under LGPLv2.1 (or later) and is also available
+ * under a commercial license.
+ *****************************************************************************
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+#ifndef MICRODNS_IP_SOURCE_H
+#define MICRODNS_IP_SOURCE_H
+
+#if defined (__unix__) || defined (__APPLE__)
+# include <arpa/inet.h>
+#elif defined(_WIN32)
+# include <ws2tcpip.h>
+#endif
+
+# ifdef __cplusplus
+extern "C" {
+# endif
+
+struct ip_source {
+    unsigned int family;
+    union {
+        struct {
+            struct in_addr addr;
+        } ipv4;
+        struct {
+            struct in6_addr addr;
+        } ipv6;
+    };
+};

```

Comment:
I would have put this new struct in the microdns.h file.

Also, ip_source is a good name for a variable for your usecase, but it's not generic enough for the name of this struct (We may want to use this same struct as a destination for example).

I propose mdns_ip.

Reply:
Sounds fair enough to me. Thanks.

Reply:
@tguillem I've made the requested changes.